MWWebPlugin/MWWebPlugin/WebStep/MWWebViewController.swift

```diff
@@ -52,13 +55,19 @@ public class MWWebViewController: MWStepViewController {
         } else {
             self.configureNavigationBar()
         }
+        if (self.hideNavigationBar) {
+            self.navigationController?.setNavigationBarHidden(true, animated: animated)
+        }
     }
     
     public override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
         if (!self.hideNavigation) {
             self.navigationController?.setToolbarHidden(true, animated: animated)
         }
+        if (self.hideNavigationBar) {
+            self.navigationController?.setNavigationBarHidden(false, animated: animated)
+        }
```

Comment:
I'm a bit worried about how this all interacts with default behaviour of unhiding navigationBar before a step appears, i.e. for the next step. See `UIViewController+Theme`.

And I also just noticed that we might be unhiding the navigationBar when the traitCollection changes in `StepNavigationViewController`, which would be an pre-existing bug.

Maybe we need to think about separating the navigationBar hidden management from the navBar theming and ensure it is handled reliably when navigating between steps?

I'll approve this for now, but if you agree, I think we should create a ticket to review this in future.

Reply:
Agree. To be able to give the step more freedom on how/when to display the navigation bar, we would need to refactor/centralise this behaviour.

Reply:
Ok, I'll create a wire task.